Merge phase reports failure when PR actually merged (post-merge cleanup error)

## Problem

The shepherd merge phase reports `[ERROR] failed to merge PR #N` and exits with code 4 (`NEEDS_INTERVENTION`) even when the PR was successfully merged. This happens because `merge-pr.sh` uses `set -euo pipefail` and any failure in post-merge cleanup (label cleanup, branch deletion, worktree removal) causes a non-zero exit, which `merge.py` treats as a merge failure without verifying actual PR state.

## Observed Output

```
[21:46:50] [OK] PR #392 approved by Judge (166s)
[21:46:50] [ERROR] No progress file found for task c8237db
...
[21:46:56] [ERROR] failed to merge PR #392
```

But `gh pr view 392 --json state` returns `MERGED` (merged at 21:46:52Z).

## Root Cause

In `merge.py:56-64`, the `except CalledProcessError` handler immediately declares failure and marks the issue as `loom:blocked` without checking whether the PR was actually merged:

```python
except subprocess.CalledProcessError:
    self._mark_issue_blocked(
        ctx, "merge_failed", f"failed to merge PR #{ctx.pr_number}"
    )
    return PhaseResult(
        status=PhaseStatus.FAILED,
        message=f"failed to merge PR #{ctx.pr_number}",
        phase_name="merge",
    )
```

The `merge-pr.sh` script has `set -euo pipefail`, so any failure in post-merge steps (lines 216-279: label cleanup, branch deletion, worktree removal) causes a non-zero exit even though the merge itself succeeded on line 152-155.

## Side Effects

Because the phase declares failure:
1. `_mark_issue_blocked` adds `loom:blocked` label to the (now-closed) issue — stale label left behind
2. A "Shepherd blocked" comment is posted on the closed issue
3. The systematic failure tracker records a spurious `merge_failed` entry
4. Shepherd exits with code 4 instead of 0

## Suggested Fix

The `validate()` method at lines 74-101 of `merge.py` already has the correct logic to check actual PR state. Reuse it in the error handler:

```python
except subprocess.CalledProcessError:
    # Check if PR merged despite script error (post-merge cleanup may have failed)
    if self.validate(ctx):
        return PhaseResult(
            status=PhaseStatus.SUCCESS,
            message=f"PR #{ctx.pr_number} merged (recovered from post-merge cleanup error)",
            phase_name="merge",
            data={"merged": True},
        )
    self._mark_issue_blocked(
        ctx, "merge_failed", f"failed to merge PR #{ctx.pr_number}"
    )
    return PhaseResult(
        status=PhaseStatus.FAILED,
        message=f"failed to merge PR #{ctx.pr_number}",
        phase_name="merge",
    )
```

## Environment

- Loom version: 0.2.3
- Repository: spheresemi/sphere
- Shepherd mode: `--merge` (force mode)
- Multiple concurrent shepherds running (issues 385, 386, 387, 389)

--- a/loom-tools/src/loom_tools/shepherd/phases/merge.py
+++ b/loom-tools/src/loom_tools/shepherd/phases/merge.py
@@ -59,6 +59,14 @@ class MergePhase:
                     phase_name="merge",
                 )
             except subprocess.CalledProcessError:
+                # Check if PR merged despite script error (post-merge cleanup may have failed)
+                if self.validate(ctx):
+                    return PhaseResult(
+                        status=PhaseStatus.SUCCESS,
+                        message=f"PR #{ctx.pr_number} merged (recovered from post-merge cleanup error)",
+                        phase_name="merge",
+                        data={"merged": True},
+                    )
                 self._mark_issue_blocked(
                     ctx, "merge_failed", f"failed to merge PR #{ctx.pr_number}"
                 )

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -5916,6 +5916,69 @@ class TestMergePhase:
         assert result.status == PhaseStatus.FAILED
         assert "merge-pr.sh not available" in result.message
 
+    @patch("loom_tools.shepherd.phases.merge.subprocess.run")
+    def test_recovers_when_pr_merged_despite_script_error(
+        self,
+        mock_subprocess: MagicMock,
+        mock_context: MagicMock,
+    ) -> None:
+        """Should return SUCCESS when PR actually merged despite CalledProcessError.
+
+        Post-merge cleanup failures (label removal, branch deletion, worktree
+        removal) cause merge-pr.sh to exit non-zero even though the merge
+        itself succeeded. The error handler should check actual PR state via
+        validate() before declaring failure (issue #2400).
+        """
+        mock_context.config = ShepherdConfig(issue=42, mode=ExecutionMode.FORCE_MERGE)
+        mock_context.check_shutdown.return_value = False
+        mock_context.pr_number = 100
+        mock_context.repo_root = Path("/fake/repo")
+        mock_context.run_script.side_effect = subprocess.CalledProcessError(1, "merge-pr.sh")
+
+        # Simulate gh pr view returning MERGED state
+        mock_subprocess.return_value = subprocess.CompletedProcess(
+            args=[], returncode=0, stdout="MERGED\n", stderr=""
+        )
+
+        merge = MergePhase()
+        result = merge.run(mock_context)
+
+        assert result.status == PhaseStatus.SUCCESS
+        assert "recovered from post-merge cleanup error" in result.message
+        assert result.data.get("merged") is True
+
+    @patch("loom_tools.common.systematic_failure.detect_systematic_failure")
+    @patch("loom_tools.common.systematic_failure.record_blocked_reason")
+    @patch("loom_tools.shepherd.phases.merge.subprocess.run")
+    def test_still_fails_when_pr_not_merged_and_script_error(
+        self,
+        mock_subprocess: MagicMock,
+        mock_record: MagicMock,
+        mock_detect: MagicMock,
+        mock_context: MagicMock,
+    ) -> None:
+        """Should still return FAILED when PR genuinely failed to merge.
+
+        When merge-pr.sh fails and the PR is NOT in MERGED state,
+        the original failure behavior should be preserved (issue #2400).
+        """
+        mock_context.config = ShepherdConfig(issue=42, mode=ExecutionMode.FORCE_MERGE)
+        mock_context.check_shutdown.return_value = False
+        mock_context.pr_number = 100
+        mock_context.repo_root = Path("/fake/repo")
+        mock_context.run_script.side_effect = subprocess.CalledProcessError(1, "merge-pr.sh")
+
+        # Simulate gh pr view returning OPEN state (merge genuinely failed)
+        mock_subprocess.return_value = subprocess.CompletedProcess(
+            args=[], returncode=0, stdout="OPEN\n", stderr=""
+        )
+
+        merge = MergePhase()
+        result = merge.run(mock_context)
+
+        assert result.status == PhaseStatus.FAILED
+        assert f"failed to merge PR #100" in result.message
+
 
 class TestPhaseStatus:
     """Test PhaseStatus enum and PhaseResult."""

